src/module-rebuilder.ts

```diff
@@ -297,16 +339,26 @@ export class ModuleRebuilder {
     const shimExt = process.env.ELECTRON_REBUILD_TESTS ? 'ts' : 'js';
     const executable = process.env.ELECTRON_REBUILD_TESTS ? path.resolve(__dirname, '..', 'node_modules', '.bin', 'ts-node') : process.execPath;
 
+    const napiVersion = await this.getNapiVersion();
+    const args = napiVersion
+      ? [
+        '--runtime=napi',
+        `--target=${napiVersion}`,
+      ]
+      : [
+        '--runtime=electron',
+        `--target=${this.rebuilder.electronVersion}`,
+      ];
+
     await spawn(
       executable,
       [
         path.resolve(__dirname, `prebuild-shim.${shimExt}`),
         prebuildInstallPath,
         `--arch=${this.rebuilder.arch}`,
         `--platform=${process.platform}`,
-        '--runtime=electron',
-        `--target=${this.rebuilder.electronVersion}`,
-        `--tag-prefix=${this.rebuilder.prebuildTagPrefix}`
+        `--tag-prefix=${this.rebuilder.prebuildTagPrefix}`,
+        ...args,
```

Comment:
```suggestion
        ...await this.getRebuildRuntimeArgs(),
```

May need parentheses, I didn't test this :sweat_smile: